api/capturesource_test.go

```diff
@@ -0,0 +1,180 @@
+		//       Add test case for this and fix issue.
+		{
+			resStatus: fiber.StatusOK,
+			reqBody: `{
+				"name": "Ochre Point",
+				"location": "-35.221389,138.46333333",
+				"camera_hardware": "rpi cam v2 - wide angle lens",
+				"site_id": 10192840284
+			}`,
+		},
+		{
+			resStatus: fiber.StatusOK,
+			reqBody: `{
+				"name": "Ochre Point",
+				"location": "-35.221389,138.46333333",
+				"camera_hardware": "rpi cam v2 - wide angle lens"
+			}`,
+		},
+		{
+			resStatus: fiber.StatusOK,
+			reqBody: `{
+				"name": "Ochre Point",
+				"location": "-35.221389,138.46333333"
+			}`,
+		},
+		{
+			resStatus: fiber.StatusBadRequest,
+			reqBody: `{
+				"name": true,
+				"location": [1,2,3],
+				"camera_hardware": 123456,
+				"site_id": false,
+				"extra": "fields"
+			}`,
+		},
+		{
+			resStatus: fiber.StatusBadRequest,
+			reqBody:   "something that is not json",
+		},
+		{
+			resStatus: fiber.StatusBadRequest,
+			reqBody:   "",
+		},
+	}
+
+	// Initialize app.
+	ds_client.Init(false)
+	app := fiber.New()
+	RegisterAPIRoutes(app)
+
+	for _, testcase := range testcases {
+		// Send test request.
+		req := httptest.NewRequest("POST", "/api/v1/capturesources", strings.NewReader(testcase.reqBody))
+		req.Header.Set("Content-Type", "application/json")
+		resp, err := app.Test(req, -1)
+
+		// Check request, response.
+		if err != nil {
+			t.Fatal(err.Error())
+		}
+
+		if resp.StatusCode != testcase.resStatus {
+			t.Log("scenario: \n", testcase.reqBody)
+			t.Fatal("expected HTTP code", testcase.resStatus, ", got", resp.StatusCode)
+		}
+	}
+}
+
+func TestGetCaptureSourcesWithLimitAndOffset(t *testing.T) {
+
+	// Initialize app.
+	ds_client.Init(false)
+	app := fiber.New()
+	RegisterAPIRoutes(app)
+
+	// Populate datastore with test data.
+	for _, body := range []string{
+		`{
+			"name": "Ochre Point",
+			"location": "-35.221389,138.46333333",
+			"camera_hardware": "rpi cam v2 - wide angle lens",
+			"site_id": 10192840284
+		}`,
+		`{
+			"name": "Carrickalinga",
+			"location": "-35.4270000,138.3148330",
+			"camera_hardware": "rpi cam v2"
+		}`,
+		`{
+			"name": "Windara",
+			"location": "-35.221389,138.46333333",
+			"camera_hardware": "rpi cam v2 - wide angle lens",
+			"site_id": 12345678910
+		}`,
+	} {
+		// Send test request.
+		req := httptest.NewRequest("POST", "/api/v1/capturesources", strings.NewReader(body))
+		req.Header.Set("Content-Type", "application/json")
+		app.Test(req, -1)
+	}
+
+	testcases := []struct {
+		limit  int
+		offset int
+	}{
+		{
+			limit: 0, offset: 0,
+		},
+		{
+			limit: 1, offset: 0,
+		},
+		{
+			limit: 2, offset: 0,
+		},
+		{
+			limit: 1, offset: 1,
+		},
+		{
+			limit: 2, offset: 1,
+		},
+	}
+
+	for _, testcase := range testcases {
+		// Send request
+		url := fmt.Sprintf("/api/v1/capturesources?limit=%d&offset=%d", testcase.limit, testcase.offset)
+		req := httptest.NewRequest("GET", url, nil)
+		resp, err := app.Test(req)
+		if err != nil {
+			t.Fatal(err.Error())
```

Comment:
You generally don't use t.Fatal for individual test cases, because this will stop execution of this Test* funciton. If you use t.Errorf, you can still go through the other tests cases in the test table.

Reply:
and ditto for the other calls to t.Fatal below.

Reply:
Using t.Errorf now